Sources/GRPCCore/GRPCServer.swift

```diff
@@ -207,40 +207,49 @@ public struct GRPCServer: Sendable {
     var listeners: [RPCAsyncSequence<Stream>] = []
     listeners.reserveCapacity(self.transports.count)
 
-    for transport in self.transports {
-      do {
-        let listener = try await transport.listen()
-        listeners.append(listener)
-      } catch let cause {
-        // Failed to start, so start stopping.
-        self.state.store(.stopping, ordering: .sequentiallyConsistent)
-        // Some listeners may have started and have streams which need closing.
-        await self.rejectRequests(listeners)
+    try await withThrowingTaskGroup(of: Void.self) { group in
+      for transport in self.transports {
+        group.addTask {
+          await transport.listen()
+        }
 
-        throw RuntimeError(
-          code: .failedToStartTransport,
-          message: """
-            Server didn't start because the '\(type(of: transport))' transport threw an error \
-            while starting.
-            """,
-          cause: cause
-        )
+        for try await event in transport.events {
```

Comment:
We'll need to do each of these in a separate task, otherwise we'll block subsequent transports from starting

Reply:
Ah, this wasn't being done in a separate task because when I first implemented, the events were only listen-related.

You're right that if we keep this a generic "transport event" stream then we need to have it run as long as listen runs. However this makes things trickier because we are appending the published accepted streams to the `listeners` array, and we don't want to use that `listeners` array until all of the transports have started and they've all published their accepted streams. If the `events` stream is long running, as well as the call to `listen()`, we can't simply wait on the group's tasks to be finished before we call `handleRequests(listeners)`.

I'm not sure how to wait for all the published accepted streams to be present - we basically need to wait for all transports to be ready, which is what we added this even stream for.